### ISSUE TITLE
Font file's missing the space character

### ISSUE THREAD

**Issue by bstoney7** [Ranks/emojione#222]
Font file doesn't display spaces. When the spacebar is pressed, nothing shows.

**Comment by mikebe11** [comment]
@bstoney7 To help out whoever may look into this issue can you provide the font you're referring to, Apple or Android? Please include the OS you're seeing this on and any other details like that which may be helpful.
Thanks

**Comment by bstoney7** [comment]
I'm referring to the Android font file, tested on Android versions 4.4.4
and 5.1 (no spaces appear)
On 4.1.1, text fails to display at all

On Tue, Mar 29, 2016 at 12:11 AM, mikebe11 <notifications@github.com> wrote:

> @bstoney7 <https://github.com/bstoney7> To help out whoever may look into
> this issue can you provide the font you're referring to, Apple or Android?
> Please include the OS you're seeing this on and any other details like that
> which may be helpful.
> Thanks
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/Ranks/emojione/issues/222#issuecomment-202619767>
>



-- 
Kwame Twum
024-971-3683

**Comment by gileichenbaum** [comment]
No white spaces shown, tested on Android .ttf file on Android 5.0.2

**Comment by miguelsousa** [comment]
The Android font has the space character. Its width is 200 font units, whereas the width of the emoji characters is 1208 font units, but it's in there.

**Comment by bstoney7** [comment]
Even when I type raw text without emojis like: How are you?, it appears as
Howareyou? the emoji characters shouldn't have anything to do with this,
should they?

<https://www.avast.com/en-us/lp-safe-emailing-2109?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail&utm_term=oa-2109-v2-a>
Virus-free.
www.avast.com
<https://www.avast.com/en-us/lp-safe-emailing-2109?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail&utm_term=oa-2109-v2-a>
<#DDB4FAA8-2DD7-40BB-A1B8-4E2AA1F9FDF2>

On Wed, Apr 20, 2016 at 11:15 AM, Miguel Sousa <notifications@github.com>
wrote:

> The Android font has the space character. Its width is 200 font units,
> whereas the width of the emoji characters is 1208 font units, but it's in
> there.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/Ranks/emojione/issues/222#issuecomment-212384542>
>



-- 
Kwame Twum
024-971-3683

**Comment by miguelsousa** [comment]
@bstoney7 no, it shouldn't. Emoji One's Android font does not contain any A thru z characters, so those, and likely also the space, must be coming from another font. Could you post a screenshot of the text? This will help to identify the font.

**Comment by bstoney7** [comment]
![text_shot](https://cloud.githubusercontent.com/assets/5204377/14764022/97b0c0b0-09a0-11e6-9b2d-94efcde9fc8e.JPG)

**Comment by miguelsousa** [comment]
The paragraph gray text looks like a stretched version of [Oxygen](https://www.google.com/fonts/specimen/Oxygen).
The rest is [Roboto](https://www.google.com/fonts/specimen/Roboto).

**Comment by bstoney7** [comment]
Yep, the rest is Roboto like you said... 
Does his mean that, in assigning EmojiOne Android font to label, the label defaults to a particular font to display text?

**Comment by miguelsousa** [comment]
I don't know.

**Comment by aat-antoine** [comment]
@gileichenbaum Thx your solution works!

**Comment by Pan-Zhang** [comment]
@aat-antoine Hi, how to solve the problem? Can you tell me the step?

**Comment by aat-antoine** [comment]
@Pan-Zhang Just as @gileichenbaum said ==> input = input.replaceAll(" ", "\u2000");

**Comment by Pan-Zhang** [comment]
@aat-antoine  Thanks, i have solve the problem. But i found some android mobile phone don't support '\u2000', so i use string.xml file to replace "\u0200", it works good.

**Comment by CriseX** [comment]
A potentially bad fix for this might actually be removing said space character from the font file.

However, this is not a one size fits all thing. I saw the same behavior when I briefly tested applying the font to a TextView, which as I understand it is actually not what you should do (since you will then be at the mercy of how Android happens to handle rendering symbols not part of the font you explicitly set), because the font is not a full font intended for displaying text to begin with.

If you want to apply the font as a typeface to specific views, the safest bet, that is never going to create a problem for you, is to merge the provided emojione font with the actual text font you want to use thus resulting into one single font file you can apply.

Alternative approach would be to specifically set the typeface for only symbols covered by the emojione font by using something like SpannableStringBuilder, thus leaving the typeface set for the rest of the contents intact (this approach might be a performance problem waiting to happen depending on your use case).
